### PR TITLE
Internal: Speed up lint process by running in concurrently

### PR DESCRIPTION
**Pull request by thangngoc89**
Before: 23s
After: 14s 

Should I apply this change to boilerplate and docs as well ?

### CODE REVIEW

**Comment by MoOx** [comment]
> Should I apply this change to boilerplate and docs as well ?

Sure, even if this will be less noticeable, since flow is probably the biggest process here.

**Comment by thangngoc89** [comment]
> since flow is probably the biggest process here.

Not true.eslint is the lowest.

**Comment by MoOx** [comment]
Did you try https://www.npmjs.com/package/npm-run-parallel ?

**Comment by thangngoc89** [comment]
> Did you try https://www.npmjs.com/package/npm-run-parallel ?

concurrently is better. Since doesn't fail if 1 process fail. 

But I'm rethink about this. Their is no color in the output. :(

**Comment by thangngoc89** [comment]
PR updated.

**Comment by MoOx** [comment]
Everything should fail if one fail.

**Comment by MoOx** [comment]
What about npm-run-all ?

**Comment by thangngoc89** [comment]
> Everything should fail if one fail.

I don't like this. This is a linting process, I prefer get all the errors and fix them all.

**Comment by MoOx** [comment]
See https://github.com/MoOx/eslint-config-i-am-meticulous/commit/45119fd78944c9cc95a64b9cc163f2ebad4dc9af

**Comment by MoOx** [comment]
Anyway, process should exit with the correct code if one fail.

**Comment by thangngoc89** [comment]
Woa. npm-run-all is cleaner. I'll go for it.

**Comment by thangngoc89** [comment]
PR updated.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ## Boilerplate
 
+- Added: Run lint process in parallel with [concurrently](https://www.npmjs.com/package/npm-run-all)
```
Comment (MoOx) [inline CHANGELOG.md]: changelog has to be updated :)

File: package.json [hunk 2]
```diff
@@ -167,7 +169,7 @@
     "lint:js:flow": "babel-node scripts/flow-check.js",
     "lint:js": "npm run lint:js:eslint && npm run lint:js:flow",
     "lint:css": "stylelint \"+(boilerplate|docs)/web_modules/**/*.css\"",
-    "lint": "npm run lint:js && npm run lint:css",
+    "lint": "npm-run-all --parallel lint:css lint:js:*",
```
Comment (MoOx) [inline package.json]: lint:* does not work ?

File: CHANGELOG.md (hunk 1 above)
Comment (thangngoc89) [inline CHANGELOG.md]: Haha. Stupid mistake.

File: package.json (hunk 2 above)
Comment (thangngoc89) [inline package.json]: It will run `lint:js:flow && lint:js:eslint && lint:js ` => flow and eslint are run twice

Comment (MoOx) [inline package.json]: so we can remove lint:js, and make the code simpler?

Comment (MoOx) [inline package.json]: I tried this on another project

```
    "lint:js:eslint": "eslint --ignore-path .gitignore --fix .",
    "lint:js:flow": "flow src",
    "lint": "npm-run-all --parallel lint:*",
```

Got

```
ERROR: Task not found: "lint:*"
```

With 

```
    "lint:js:eslint": "eslint --ignore-path .gitignore --fix .",
    "lint:js:flow": "flow src",
    "lint:js": "npm-run-all --parallel lint:js:*",
    "lint": "npm-run-all --parallel lint:*",
```

Not duplicate output. So are you sure?

Comment (thangngoc89) [inline package.json]: Sorry. I was wrong. It runs `lint:js && lint:css`. It treats the colon as sperator so you will need to do this: "npm-run-all lint:**"

**Comment by MoOx** [comment]
FYI https://github.com/eslint/eslint/issues/5893

**Comment by thangngoc89** [comment]
:+1: That's very interesting

**Comment by MoOx** [comment]
Now this take around 5 to 7s.

**Pull request merged**